File: find-the-difference-of-two-arrays/find-the-difference-of-two-arrays.java
class Solution {
    public List<List<Integer>> findDifference(int[] nums1, int[] nums2) {
        return Arrays.asList(getElementsOnlyInFirstList(nums1, nums2), getElementsOnlyInFirstList(nums2, nums1));
    }

    private List<Integer> getElementsOnlyInFirstList(int[] nums1, int[] nums2) {
        Set<Integer> onlyInNums1 = new HashSet<>();
        Set<Integer> set2 = new HashSet<>();
        for (int num : nums2) {
            set2.add(num);
        }
        for (int num : nums1) {
            if(!set2.contains(num)) {
                onlyInNums1.add(num);
            }
        }
        return new ArrayList<>(onlyInNums1);
    }
}

File: 1392-find-the-difference-of-two-arrays/find-the-difference-of-two-arrays.py
class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        map1 = Counter(nums1)
        map2 = Counter(nums2)
        list1 = set()
        list2 = set()
        for num in nums1:
            if num not in map2.keys() and num not in list1:
                list1.add(num)
        for num in nums2:
            if num not in map1.keys() and num not in list2:
                list2.add(num)
        return [list(list1), list(list2)]